types/evidence.go

```diff
@@ -229,6 +234,7 @@ func (l *LightClientAttackEvidence) ABCI() []abci.Evidence {
 			Height:           l.Height(),
 			Time:             l.Timestamp,
 			TotalVotingPower: l.TotalVotingPower,
+			VotingPower:      0,
```

Comment:
Why is `VotingPower` 0 for the case of `LightClientAttackEvidence`?

Reply:
This is because the `LightClientAttackEvidence` `l` has neither `ValidatorPower` nor `VotingPower`. The reason is that the LightClient that detects the *attack* doesn't have any address or VotingPower.
Here I focus on the newly added field and explicitly set it to `0`.

Reply:
`LightClientAttackEvidence` has `ByzantineValidators []*Validator` and each validator set into `abci.Evidence.Validator`. I think here is `abci.Evidence.Validator.Power` means `ValidatorPower`. Should we set `val.VotingPower` into `abci.Evidence.VotingPower`?

Reply:
Ah, indeed, if we want to set `VotingPower`, that's the field.

Reply:
I think we should put `val.VotingPower` instead of 0 int the field `abci.Evidence.VotingPower`.
```
for idx, val := range l.ByzantineValidators {
		pb := abci.Validator{
			Address: val.Address,
			Power:   val.StakingPower,
		}
		abciEv[idx] = abci.Evidence{
			Type:             abci.EvidenceType_LIGHT_CLIENT_ATTACK,
			Validator:        pb,
			Height:           l.Height(),
			Time:             l.Timestamp,
			TotalVotingPower: l.TotalVotingPower,
		}
	}
```
As you can see in the code above, each of the byzantine validators can be seen creating an `abci.Evidence`. We have created this field to turn over the voting power of the Byzantine validator, so I think we should turn over this value.

Reply:
Oh, I found that you had added `VotingPower` to `abci.Validator`. If so, you might remove the `VotingPower` field from other abci structures including `abci.Validator`, for example, `abci,VoteInfo`, `abci.Evidence`.

Reply:
You're right. The `VotingPower` currently associated with `abci.Evidence` should be correctly associated with the `abci.Validator` and I'll fix as such.